src/cmap/connection_pool.ts

```diff
@@ -252,7 +252,7 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
     this[kLogger] = new Logger('ConnectionPool');
     this[kConnections] = new Denque();
     this[kPending] = 0;
-    this[kCheckedOut] = 0;
+    this[kCheckedOut] = new Set();
```

Comment:
Since this is of type `Set`, I'm just curious how equality is defined for connections.  Is it just reference equality?

Reply:
The `Set` will use referential equality for the connections, yes.